security_privacy_ln.asciidoc

```diff
@@ -0,0 +1,614 @@
+
+In the past, the Bitcoin network faced an attempted denial-of-service attack where attackers spammed the network with low-fee transactions.
+Many of these transactions were not selected by miners due to their low transaction fees, so the attackers could consume network resources without paying the fees.
+To address this issue, a minimum transaction relay fee that set a threshold fee that nodes require to propagate transactions was set.
+This measure largely ensured that the transactions that consume network resources will eventually pay their chain fees.
+The minimum relay fee is acceptable to regular users but would hurt attackers financially if they tried to spam the network.
+While some transactions may not make it into valid blocks during high fee environments, these measures have largely been effective at disincentivizing this type of spam.
+
+=== DoS in Lightning
+
+Similarly to Bitcoin, the Lightning Network charges fees for the use of its public resources, but in this case, the resources are public channels, and the fees come in the form of routing fees.
+The ability to route payments through nodes in exchange for fees provides the network with a large scalability benefit - nodes that are not directly connected can still transact - but it comes at the cost of exposing a public resource that must be protected against DoS attacks.
+
+When a Lightning node forwards a payment on your behalf, it uses bandwidth to update its commitment transaction, and the amount of the payment is reserved in their channel balance until it is settled or failed.
+In successful payments, this is acceptable because the node is eventually paid out its fees.
+Failed payments do not incur fees in the current protocol.
+This allows nodes to costlessly route failed payments through any channels.
+This is great for legitimate users, who wouldn't like to pay for failed attempts, but also allows attackers to costlessly consume nodes' resources - much like the low-fee transactions on Bitcoin that never end up paying miner fees.
+
+At the time of writing, a discussion is https://lists.linuxfoundation.org/pipermail/lightning-dev/2020-June/002734.html[ongoing] on the lightning-dev mailing list as to how best address this issue.
+
+=== Known DoS attacks
+
+There are two known DoS attacks on public Lightning channels which render a target channel, or a set of target channels, unusable.
+Both attacks involve routing payments through a public channel, then holding them until their timeout, thus maximizing the attack's duration.
+The requirement to fail payments to not pay fees is fairly simple to meet because malicious nodes can simply reroute payments to themselves.
+In the absence of fees for failed payments, the only cost to the attacker is the on-chain cost of opening a channel to dispatch these payments through, which can be trivial in low fee environments.
```

Comment:
The cost also includes worst case closing 2x the channels (if they're doing it via circular routes), and also the opportunity cost of the Bitcoin they need to put into channels to launch the attack.